File: testPrograms/tc.cc
#include <stdlib.h>

#include <string>
#include <iostream>

using namespace std;

#include "pluton/client.h"


int
main(int argc, char** argv)
{
  pluton::client        C("tc");
  pluton::clientRequest R1;
  pluton::clientRequest R2;
  pluton::clientRequest R3;

  cout << C.getAPIVersion() << endl;
  if (!C.initialize("./lookup.map")) {
    cout << C.getFault().getMessage("Fatal: pluton::initialize() failed")  << endl;
    exit(1);
  }

  R1.setRequestData("This is the first request");
  R2.setRequestData("This is the second request");
  R3.setRequestData("This is the third");

  C.setTimeoutMilliSeconds(2000);

  C.addRequest("Mail.getFolder.XML.1", &R1);
  C.addRequest("AB.addEvent.PHP.0", &R2);
  C.addRequest("AB.addEvent.PHP0", &R3);
  if (C.executeAndWaitAll() <= 0) {
    cout << "execute failed" << endl;
  }

  cout << "R1 Fault Code=" << R1.getFaultCode() << " Text=" << R1.getFaultText() << endl;
  cout << "R2 Fault Code=" << R2.getFaultCode() << " Text=" << R2.getFaultText() << endl;
  cout << "R3 Fault Code=" << R3.getFaultCode() << " Text=" << R3.getFaultText() << endl;

  string s;

  cin >> s;             // Block until input

  return 0;
}

File: wrappers/perl.old/tc.pl
#! /usr/local/bin/perl -w

use Yahoo::pluton;

print Yahoo::pluton::client::getAPIVersion(), "\n";

my $c = Yahoo::pluton::client->new("perlbaby");
$c->initialize("");
# $c->setDebug(1);

my $r = Yahoo::pluton::clientRequest->new();

my $rd = "123456789A123456789A12345X";
$r->setRequestData($rd);

print "Request set\n";

if (! $c->addRequest("system.echo.0.raw", $r)) {
    print $c->getFault()->getMessage("add Request"), "\n";
    exit(1);
}

my $newR = $r;
my $newC = $c;

$c->executeAndWaitAll();
if ($c->hasFault()) {
    print $c->getFault()->getMessage("E&W"), "\n";
    exit(1);
}

my $s = $r->getResponseData();

print "Response Data: ", $s, "\n";
